### PR TITLE
Add Compose SDK

### PR DESCRIPTION
**Pull request by filbabic**
### 🎯 Goal

We're adding a Jetpack Compose version of the SDK, that features our well known experience and features.

The goal is to provide a modern set of tools and components, that let Compose users utilize our SDK.

### 🛠 Implementation details

All the SDK code is featured in `stream-chat-android-compose`, while the sample app lives in `stream-chat-android-compose-sample`.

Components are split up in logical structures and packages, based on which screens they're used in (e.g. Channels, Messages..).

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] ~New code is covered by unit tests~
- [ ] ~Comparison screenshots added for visual changes~
- [x] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### 🎉 GIF

giphy

### CODE REVIEW

File: buildSrc/src/main/kotlin/io/getstream/chat/android/Configuration.kt [hunk 1]
```diff
@@ -1,7 +1,7 @@
-package com.getstream.sdk.chat
+package io.getstream.chat.android
 
 object Configuration {
-    const val compileSdkVersion = 29
+    const val compileSdkVersion = 30
```
Comment (zsmb13) [inline buildSrc/src/main/kotlin/io/getstream/chat/android/Configuration.kt:4]: Required for Compose support

File: stream-chat-android/build.gradle [hunk 2]
```diff
@@ -65,7 +65,6 @@ dependencies {
     implementation Dependencies.androidLegacySupport
     implementation Dependencies.androidxLifecycleExtensions
     implementation Dependencies.androidxLifecycleViewModelKtx
-    implementation Dependencies.androidxMedia
```
Comment (zsmb13) [inline stream-chat-android/build.gradle:68]: This was unused as far as I understand

File: buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt [hunk 3]
```diff
@@ -0,0 +1,210 @@
+@file:Suppress("RedundantVisibilityModifier")
+
+package io.getstream.chat.android
```
Comment (zsmb13) [inline buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt:3]: Moved to the proper package just for consistency. Also grouped versions inside a `Versions` object so that they can be reasonably referenced in Gradle scripts when needed. Only exposed Kotlin and Compose versions for now.

File: docusaurus/docs/Android/01-basics/03-getting-started.md [hunk 4]
```diff
@@ -84,10 +84,12 @@ To learn about how to create a token and different user types, see [Tokens & Aut
 
 If the `connectUser` call was successful, you are now ready to use the SDK! 🎉
 
-The methods of the `ChatClient` class allow you to create channels, send messages, add reactions, and perform many more low-level operations. You can also use the SDK's pre-built UI Components that come with ViewModels that will perform data fetching and sending for you, as described below.
+The methods of the `ChatClient` class allow you to create channels, send messages, add reactions, and perform many more low-level operations. You can also use the SDK's pre-built UI Components that will perform data fetching and sending for you, as described below.
 
 ### Adding UI Components
 
-To add the UI Components to your application, see the [Get Started](../03-ui/01-overview.md) page of the UI Components section first, which explains how the components work.
+There are two UI Component implementations available: one built on regular old Android Views that you can use in XML layouts, and another built from the ground up in [Jetpack Compose](https://developer.android.com/jetpack/compose).
```
Comment (adasiewiczr) [inline docusaurus/docs/Android/01-basics/03-getting-started.md:91]: > old Android Views that you can use in XML layouts

Isn't it possible to use Jetpack Compose in XML layouts?
I would rephrase this sentence a little bit: `There are two UI Component implementations available: one built on regular old, XML based Android Views,`

**Review by adasiewiczr — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -41,7 +43,13 @@ This SDK consists of three artifacts you can build on:
     - [Sample app](/stream-chat-android-ui-components-sample)
     - [Documentation](https://getstream.io/chat/docs/sdk/android/ui/overview/)
 
-Learn more by visiting [the documentation](https://getstream.io/chat/docs/sdk/android/).
+We also support Jetpack Compose! You can use our Compose UI Components as an alternative to the **UI Components**.
```
[the remark this answers is not in the document]
Comment (zsmb13) [reply README.md:46]: We mention it later down in the same README (and also on various other pages), that felt like enough.

[the remark this answers is not in the document]
Comment (adasiewiczr) [reply README.md:46]: It was a little bit weird that we mention that inside `🏗️ Jetpack Compose Sample App` paragraph but not here. Feel free to skip this comment

File: docusaurus/docs/Android/01-basics/03-getting-started.md (hunk 4 above)
Comment (zsmb13) [reply docusaurus/docs/Android/01-basics/03-getting-started.md]: Good call, updated this wording
